src/unitxt/formats.py

```diff
@@ -273,19 +312,82 @@ def process(
+
+        last_user_content = self.to_content(source)
+
+        messages.extend([{"role": "user", "content": last_user_content}])
+
+        return messages
+
+    def process(
+        self, instance: Dict[str, Any], stream_name: Optional[str] = None
+    ) -> Dict[str, Any]:
+        chat = self.to_chat(instance, stream_name)
+        instance.pop("target_prefix")
+        instance["source"] = json.dumps(chat)
+        return instance
+
+
+class HFSystemFormat(OpenAIFormat):
+    r"""Formats the complete input for the model using the HuggingFace chat template of a given model.
+
+    HFSystemFormat expects the input instance to contain:
+    1. A field named "system_prompt" whose value is a string (potentially empty) that delivers a task-independent opening text.
+    2. A field named "source" whose value is a string verbalizing the original values in the instance (as read
+    from the source dataset), in the context of the underlying task.
+    3. A field named "instruction" that contains a (non-None) string.
+    4. A field named with the value in arg 'demos_field', containing a list of dicts, each dict with fields "source"
+    and "target", representing a single demo.
+    5. A field named "target_prefix" that contains a string to prefix the target in each demo, and to end the whole generated prompt.
+
+    SystemFormat formats the above fields into a single string to be inputted to the model. This string overwrites
+    field "source" of the instance.
+
+    Example:
+        HFSystemFormat(model_name="HuggingFaceH4/zephyr-7b-beta")
+
+        Uses the template defined the in tokenizer_config.json of the model:
+
+        "chat_template": "{% for message in messages %}\n{% if message['role'] == 'user' %}\n{{ '<|user|>\n' + message['content'] + eos_token }}\n{% elif message['role'] == 'system' %}\n{{ '<|system|>\n' + message['content'] + eos_token }}\n{% elif message['role'] == 'assistant' %}\n{{ '<|assistant|>\n'  + message['content'] + eos_token }}\n{% endif %}\n{% if loop.last and add_generation_prompt %}\n{{ '<|assistant|>' }}\n{% endif %}\n{% endfor %}",
+
+        See more details in https://huggingface.co/docs/transformers/main/en/chat_templating
+
+    """
+
+    model_name: str
+    _requirements_list = ["transformers", "Jinja2"]
+
+    def prepare(self):
+        super().prepare()
+        from transformers import AutoTokenizer
+
+        self.tokenizer = AutoTokenizer.from_pretrained(self.model_name)
+
+    def process(
+        self, instance: Dict[str, Any], stream_name: Optional[str] = None
+    ) -> Dict[str, Any]:
+        chat = self.to_chat(instance, stream_name)
+
+        target_prefix = self._retrieve_field_and_pop_from_instance(
+            instance=instance, field_name="target_prefix"
+        )
+
+        instance["source"] = (
+            self.tokenizer.apply_chat_template(
+                chat, tokenize=False, add_generation_prompt=True
+            )
+            + target_prefix
         )
```

Comment:
I don't understand why we need this.
Normally we won't be able to have a string "source" string. 
The source should be a dictionary in chat format, no?

Reply:
How do you envision the usage of it?
I envisioned it that the dictionary is in the source, and we just pass this to the model

Reply:
Basically the user and unitxt won't know and cannot know what is the format. It only know that the backend somehow handle this

Reply:
This is just for backward compatibility (for pepole who's using unitxt to get data with hf chat template in a textual format).

Reply:
Separate it to a different mechanism imo. We don't need to mix stuff that are not related from design pov